src/components/Hoverable/ActiveHoverable.tsx

```diff
@@ -83,7 +84,7 @@ function ActiveHoverable({onHoverIn, onHoverOut, shouldHandleScroll, children}:
         return () => document.removeEventListener('visibilitychange', unsetHoveredWhenDocumentIsHidden);
     }, []);
 
-    const child = useMemo(() => (typeof children === 'function' ? children(!isScrollingRef.current && isHovered) : children), [children, isHovered]);
+    const child = useMemo(() => getFirstValue(getReturnValue(children, !isScrollingRef.current && isHovered)), [children, isHovered]);
```

Comment:
I'm a bit confused, why are we fine with just taking the first value blindly?

Reply:
According to the information from @bernhardoj [here](https://github.com/Expensify/App/issues/32096#issuecomment-1842089033), the children is

 - regular single child
 - single child wrapped in the array

Thus what the function does is either takes the child directly or extract first item (child) of the array.

What exactly do you mean by _blindly_?

Reply:
CC @r8

Reply:
CC @cubuspl42

Reply:
I'm sorry, but I don't think this is going in the right direction. What is the scale of the potential change of actually fixing this situation, including the use-site? How many usages are we talking about?

Reply:
@cubuspl42 I will go through the codebase and gather information about usages 👍

Reply:
Reported back [here](https://github.com/Expensify/App/pull/33257)